cvmfs/catalog_mgr_rw.cc

```diff
@@ -1128,15 +1128,16 @@ WritableCatalogManager::SnapshotCatalogsSerialized(
     // Compress and upload catalog
     shash::Any hash_catalog(spooler_->GetHashAlgorithm(),
                             shash::kSuffixCatalog);
+    const std::string temp_compressed_path = (*i)->database_path() + ".compressed";
     if (!zlib::CompressPath2Path((*i)->database_path(),
```

Comment:
There might be a CompressPath2Null function in compression.cc, which avoids writing the temporary file.